tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks/UsageReport/WritePackageUsageData.cs

```diff
@@ -180,28 +180,37 @@ public override bool Execute()
                 assetFiles,
                 assetFile =>
                 {
-                    var properties = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+                    JObject jObj;
 
                     using (var file = File.OpenRead(Path.Combine(RootDir, assetFile)))
                     using (var reader = new StreamReader(file))
                     using (var jsonReader = new JsonTextReader(reader))
                     {
-                        while (jsonReader.Read())
-                        {
-                            if (jsonReader.TokenType == JsonToken.PropertyName &&
-                                jsonReader.Value is string value)
-                            {
-                                properties.Add(value);
-                            }
-                        }
+                        jObj = (JObject)JToken.ReadFrom(jsonReader);
                     }
 
+                    var properties = new HashSet<string>(
+                        jObj.SelectToken("$.targets.*").Select(t => ((JProperty)t).Name).Union(
+                            jObj.SelectToken("$.libraries").Select(t => ((JProperty)t).Name)),
```

Comment:
`Concat` earlier then take `Distinct` to factor out `Select`?

```C#
jObj.SelectToken("$.targets.*")
    .Concat(jObj.SelectToken("$.libraries"))
    .Select(t => ((JProperty)t).Name))
    .Distinct()
```

Thought about also suggesting `OfType` for safety, but actually I think it's good to get CI failures if this fails to make sure our prebuilt reports weren't invalidated by some NuGet change.